lib/interactive_logger/step.rb

```diff
@@ -60,9 +61,20 @@ def prefix(str)
     end
 
     def print_msg(str)
+      str = terminal_trim(str)
       blank
       @last_print_msg = str
       print str
     end
+
+    # Trim string to current terminal width and break at newline so it can be
+    # replaced rather than causing the same line to print multiple times.
+    def terminal_trim(str)
+      terminal_width = @thor_shell.terminal_width
```

Comment:
Oh, wow, this is actually way harder than I thought. We have to not count color codes, and truncate in what is possibly inside of a changed color state. Wheee....